src/library/scala/util/Using.scala

```diff
@@ -122,21 +136,37 @@ object Using {
     */
   def apply[R](resource: => R): Using[R] = new Using(resource)
 
+  private def preferentiallySuppress(primary: Throwable, secondary: Throwable): Throwable = {
+    def score(t: Throwable): Int = t match {
+      case _: VirtualMachineError                   => 4
+      case _: LinkageError                          => 3
+      case _: InterruptedException | _: ThreadDeath => 2
+      case _: ControlThrowable                      => 0
+      case e if !NonFatal(e)                        => 1 // in case this method gets out of sync with NonFatal
+      case _                                        => -1
+    }
```

Comment:
This can be optimised to perform only the minimum number of type checks, but I'm not sure if the hit to readability is worth it.

```scala
  private def preferentiallySuppress(primary: Throwable, secondary: Throwable): Throwable = {
    @inline def suppress(t: Throwable, suppressed: Throwable): Throwable = { t.addSuppressed(suppressed); t }

    if (primary.isInstanceOf[VirtualMachineError]) suppress(primary, secondary)
    else if (secondary.isInstanceOf[VirtualMachineError]) suppress(secondary, primary)
    else if (primary.isInstanceOf[LinkageError]) suppress(primary, secondary)
    else if (secondary.isInstanceOf[LinkageError]) suppress(secondary, primary)
    else if (primary.isInstanceOf[InterruptedException] || primary.isInstanceOf[ThreadDeath]) suppress(primary, secondary)
    else if (secondary.isInstanceOf[InterruptedException] || secondary.isInstanceOf[ThreadDeath]) suppress(secondary, primary)
    else {
      val primaryControlThrowable = primary.isInstanceOf[ControlThrowable]
      if (!NonFatal(primary) && !primaryControlThrowable) suppress(primary, secondary)
      else {
        val secondaryControlThrowable = secondary.isInstanceOf[ControlThrowable]
        if (!NonFatal(secondary) && ! secondaryControlThrowable) suppress(secondary, primary)
        else if (primaryControlThrowable) primary     // skip suppression to
        else if (secondaryControlThrowable) secondary // prevent memory leakage
        else suppress(primary, secondary)
      }
    }
  }
```

Reply:
not perfectly optimal - don't need to evaluate `NonFatal(x)` if `xControlThrowable` is true

Reply:
Let's... not. (and cross that bridge if we get to it)